test/dynamo/test_graph_break_messages.py

```diff
@@ -0,0 +1,221 @@
+Tensor.item
+
+from user code:
+   File "test_graph_break_messages.py", line N, in fn
+    return x.item()""",
+        )
+
+    def test_data_dependent_operator2(self):
+        def fn(x):
+            return torch.equal(x, x)
+
+        with torch._dynamo.config.patch(capture_scalar_outputs=True):
+            self.assertExpectedInlineMunged(
+                Unsupported,
+                lambda: torch.compile(fn, backend="eager", fullgraph=True)(torch.ones(3)),
+                """\
+Data dependent operator
+  Explanation: Operator `aten.equal.default` has a non-Tensor output whose value is dependent on the data of Tensor inputs.
+  Hint: Consider wrapping the operator into a PyTorch-understood custom operator (see https:/pytorch.org/tutorials/advanced/custom_ops_landing_page.html)
+
+  Developer debug context: aten.equal.default
+
+
+from user code:
+   File "test_graph_break_messages.py", line N, in fn
+    _helper(fn, (torch.ones(3),)),""",
+            )
+
+    def test_super_call_method(self):
+        def fn(it):
+            return [x for x in it]
+
+        self.assertExpectedInlineMunged(
+            Unsupported,
+            lambda: torch.compile(fn, backend="eager", fullgraph=True)(zip(range(5), range(10))),
+            """\
+Unsupported method call
+  Explanation: Dynamo does not know how to trace method `__iter__` of class `zip`
+  Hint: Avoid calling `zip.__iter__` in your code.
+  Hint: Please report an issue to PyTorch.
+
+  Developer debug context: call_method UserDefinedObjectVariable(zip) __iter__ () {}
+
+
+from user code:
+   File "test_graph_break_messages.py", line N, in fn
+    return [x for x in it]""",
+        )
+
+    def test_super_call_function(self):
+        def fn(it):
+            return [x for x in it()]
+
+        self.assertExpectedInlineMunged(
+            Unsupported,
+            lambda: torch.compile(fn, backend="eager", fullgraph=True)(zip(range(5), range(10))),
+            """\
+Unsupported function call
+  Explanation: Dynamo does not know how to trace the function `UserDefinedObjectVariable(zip)`
+  Hint: Avoid calling `UserDefinedObjectVariable(zip)` in your code.
+  Hint: Please report an issue to PyTorch.
+
+  Developer debug context: call_function UserDefinedObjectVariable(zip) [] {}
+
+
+from user code:
+   File "test_graph_break_messages.py", line N, in fn
+    return [x for x in it()]""",
+        )
+
+    def test_unsupported_context(self):
+        def fn(obj):
+            with obj:
+                return x
+
+        self.assertExpectedInlineMunged(
+            Unsupported,
+            lambda: torch.compile(fn, backend="eager", fullgraph=True)(3),
+            """\
+Unsupported context manager
+  Explanation: Dynamo does not know how to enter a `int` context manager.
+  Hint: Avoid using the unsupported context manager.
+  Hint: File an issue to PyTorch. Simple context managers can potentially be supported, but note that context managers can't be supported in general
```

Comment:
mm this is where Ed's wiki idea makes sense. We have some constraints over what kind of context managers work with Dynamo documenting this for the user would be great.